lib/rubocop/config_obsoletion.rb

```diff
@@ -50,7 +50,7 @@ def reject_obsolete!
     # Default rules for obsoletions are in config/obsoletion.yml
     # Additional rules files can be added with `RuboCop::ConfigObsoletion.files << filename`
     def load_rules # rubocop:disable Metrics/AbcSize
-      rules = LOAD_RULES_CACHE[self.class.files] ||=
+      rules = LOAD_RULES_CACHE[self.class.files.hash] ||=
```

Comment:
I did `self.class.files.count` as the key first since I expect this to only ever be added to but this seems safer.